From: Borislav Petkov <bp@suse.de>

Doing randbuilds fails like this here at link time:

  arch/x86/built-in.o: In function `arch_setup_additional_pages':
  /home/boris/kernel/linux/arch/x86/entry/vdso/vma.c:174: undefined reference to `pvclock_pvti_cpu0_va'
  make: *** [vmlinux] Error 1

This is because pvclock_pvti_cpu0_va() is defined in
arch/x86/kernel/kvmclock.c as an accessor to the private hv_clock thing
but if CONFIG_KVM_GUEST is not set, we don't have that function to link
to.

So add CONFIG_KVM_GUEST to the ifdeffery.

Signed-off-by: Borislav Petkov <bp@suse.de>
---
 arch/x86/include/asm/pvclock.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/x86/include/asm/pvclock.h b/arch/x86/include/asm/pvclock.h
index 66df22b2e0c9..1af6bee77fb0 100644
--- a/arch/x86/include/asm/pvclock.h
+++ b/arch/x86/include/asm/pvclock.h
@@ -4,7 +4,7 @@ 
 #include <linux/clocksource.h>
 #include <asm/pvclock-abi.h>
 
-#ifdef CONFIG_PARAVIRT_CLOCK
+#if defined(CONFIG_PARAVIRT_CLOCK) && defined(CONFIG_KVM_GUEST)
 extern struct pvclock_vsyscall_time_info *pvclock_pvti_cpu0_va(void);
 #else
 static inline struct pvclock_vsyscall_time_info *pvclock_pvti_cpu0_va(void)
